executor/join.go

```diff
@@ -529,17 +564,46 @@ func (e *NestedLoopApplyExec) prepare(goCtx goctx.Context) error {
 	}
 }
 
-func (e *NestedLoopApplyExec) doJoin(bigRow Row, match bool) ([]Row, error) {
+// fetchAllInners reads all data from the inner table and stores them in a slice.
+func (e *NestedLoopApplyExec) fetchAllInners(goCtx goctx.Context) error {
+	err := e.innerExec.Open(goCtx)
```

Comment:
Why we use `e.innerExec.Open(goCtx.TODO())` in `prepare()`  but use `e.innerExec.Open(goCtx)` there? And the position of `defer` is also different.

Reply:
I just don't want to change the old code as it will be removed soon.